File: android/media/src/main/java/com/dreampany/media/data/source/memory/ApkMemoryDataSource.java

package com.dreampany.media.data.source.memory;

import com.dreampany.media.data.model.Apk;
import com.dreampany.media.data.source.api.MediaDataSource;

import java.util.List;

import javax.inject.Singleton;

import io.reactivex.Maybe;


/**
 * Created by Hawladar Roman on 7/17/2018.
 * BJIT Group
 * hawladar.roman@bjitgroup.com
 */

@Singleton
public class ApkMemoryDataSource implements MediaDataSource<Apk> {

    private final ApkProvider provider;

    public ApkMemoryDataSource(ApkProvider provider) {
        this.provider = provider;
    }

    @Override
    public MediaDataSource<Apk> getThis() {
        return this;
    }

    @Override
    public boolean isEmpty() {
        return false;
    }

    @Override
    public Maybe<Boolean> isEmptyRx() {
        return null;
    }


    @Override
    public int getCount() {
        return 0;
    }

    @Override
    public Maybe<Integer> getCountRx() {
        return null;
    }

    @Override
    public boolean isExists(Apk apk) {
        return false;
    }

    @Override
    public Maybe<Boolean> isExistsRx(Apk apk) {
        return null;
    }

    @Override
    public long putItem(Apk apk) {
        return 0;
    }

    @Override
    public Maybe<Long> putItemRx(Apk apk) {
        return null;
    }

    @Override
    public List<Long> putItems(List<Apk> apks) {
        return null;
    }

    @Override
    public Maybe<List<Long>> putItemsRx(List<Apk> apks) {
        return null;
    }

    @Override
    public Apk getItem(long id) {
        return null;
    }

    @Override
    public Maybe<Apk> getItemRx(long id) {
        return null;
    }

    @Override
    public List<Apk> getItems() {
        return provider.getItems();
    }

    @Override
    public Maybe<List<Apk>> getItemsRx() {
        return provider.getItemsRx();
    }

    @Override
    public List<Apk> getItems(int limit) {
        return provider.getItems(limit);
    }

    @Override
    public Maybe<List<Apk>> getItemsRx(int limit) {
        return provider.getItemsRx(limit);
    }
}

File: android/nearby/src/main/java/com/dreampany/nearby/data/source/memory/ApkMemoryDataSource.kt
package com.dreampany.nearby.data.source.memory

import com.dreampany.nearby.data.model.media.Apk
import com.dreampany.nearby.data.source.api.ApkDataSource
import com.dreampany.nearby.data.source.mapper.ApkMapper

/**
 * Created by roman on 28/6/20
 * Copyright (c) 2020 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class ApkMemoryDataSource(
    private val mapper: ApkMapper,
    private val provider: ApkProvider
) : ApkDataSource {

    override suspend fun isFavorite(input: Apk): Boolean {
        TODO("Not yet implemented")
    }

    override suspend fun toggleFavorite(input: Apk): Boolean {
        TODO("Not yet implemented")
    }

    override suspend fun getFavorites(): List<Apk>? {
        TODO("Not yet implemented")
    }

    override suspend fun put(input: Apk): Long {
        TODO("Not yet implemented")
    }

    override suspend fun put(inputs: List<Apk>): List<Long>? {
        TODO("Not yet implemented")
    }

    override suspend fun get(id: String): Apk? {
        TODO("Not yet implemented")
    }

    override suspend fun gets(): List<Apk>? =
        provider.gets()

    override suspend fun gets(ids: List<String>): List<Apk>? {
        TODO("Not yet implemented")
    }

    override suspend fun gets(offset: Long, limit: Long): List<Apk>? {
        TODO("Not yet implemented")
    }
}